File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) 
    {
        int temp[20000000] = {0};
        int count =0, prefixSum = 0;
        temp[10000000] = 1;
        for(int i = 0; i < nums.size(); i++)
        {
            prefixSum += nums[i];
            count += temp[prefixSum - k +10000000];
            temp[prefixSum + 10000000] += 1;
        }
        return count;
    }
};

File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        
        dic = {0 : 1}
        ans = 0
        prefixSum = 0
        for i, n in enumerate(nums):
            prefixSum += n
            if prefixSum - k in dic:
                ans += dic[prefixSum - k]
            dic[prefixSum] = dic.get(prefixSum, 0) + 1
        return ans

        